server/apps/archive/archive_rewrite.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from flask import request
+
+from superdesk import get_resource_service, Service
+from superdesk.resource import Resource, build_custom_hateoas
+from apps.archive.common import item_url, CUSTOM_HATEOAS
+from superdesk.workflow import is_workflow_state_transition_valid
+from superdesk.errors import SuperdeskApiError, InvalidStateTransitionError
+from apps.packages.takes_package_service import TakesPackageService
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ArchiveRewriteResource(Resource):
+    endpoint_name = 'archive_rewrite'
+    resource_title = endpoint_name
+
+    schema = {
+        'desk_id': {"type": "string"}
+    }
+
+    url = 'archive/<{0}:original_id>/rewrite'.format(item_url)
+
+    resource_methods = ['POST']
+    privileges = {'POST': 'rewrite'}
+
+
+class ArchiveRewriteService(Service):
+    def create(self, docs, **kwargs):
+        original_id = request.view_args['original_id']
+        doc = docs[0]
+        desk_id = doc.get('desk_id')
+        archive_service = get_resource_service('archive')
+        original = archive_service.find_one(req=None, _id=original_id)
+        digital = TakesPackageService().get_take_package(original)
+
+        self._validate_rewrite(original)
```

Comment:
I think this validation should occur even before we fetch digital version as we are throwing `Cannot find article` error.